File: fix_ui.js
import fs from 'fs';
let code = fs.readFileSync('src/App.tsx', 'utf8');

const updatedInstructions = `
        {/* Instructions Modal */}
        {showInstructions && (
          <div className="absolute inset-0 bg-black/90 backdrop-blur-xl flex items-center justify-center z-[100]">
            <div className="bg-[#141417] p-8 md:p-12 rounded-[2.5rem] border border-white/10 shadow-[0_0_50px_rgba(0,0,0,0.5)] max-w-2xl w-full text-white transform transition-all">
              <div className="text-center mb-10">
                 <h2 className="text-4xl font-black text-transparent bg-clip-text bg-gradient-to-r from-blue-400 to-emerald-400 mb-3 tracking-tight">Welcome to AI Sandbox</h2>
                 <p className="text-neutral-400">Master your controls to survive and thrive.</p>
              </div>
              
              <div className="grid grid-cols-1 md:grid-cols-2 gap-8 mb-10">
                 <div className="bg-[#0A0A0B]/80 p-6 rounded-3xl border border-white/5">
                    <h3 className="text-sm uppercase tracking-widest text-neutral-500 font-bold mb-4">Movement & Combat</h3>
                    <ul className="space-y-4">
                      <li className="flex items-center gap-3"><span className="w-8 h-8 rounded-lg bg-white/10 flex items-center justify-center text-xs font-bold text-white border border-white/5">WASD</span> <span className="text-sm text-neutral-300">Move & Jump</span></li>
                      <li className="flex items-center gap-3"><span className="w-8 h-8 rounded-lg bg-white/10 flex items-center justify-center text-xs font-bold text-white border border-white/5"><svg className="w-4 h-4" fill="none" viewBox="0 0 24 24" stroke="currentColor"><path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M15 15l-2 5L9 9l11 4-5 2zm0 0l5 5M7.188 2.239l.777 2.897M5.136 7.965l-2.898-.777M13.95 4.05l-2.122 2.122m-5.657 5.656l-2.12 2.122" /></svg></span> <span className="text-sm text-neutral-300">Left Click to Mine/Attack</span></li>
                      <li className="flex items-center gap-3"><span className="w-8 h-8 rounded-lg bg-white/10 flex items-center justify-center text-xs font-bold text-white border border-white/5"><svg className="w-4 h-4" fill="none" viewBox="0 0 24 24" stroke="currentColor"><path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M15 15l-2 5L9 9l11 4-5 2zm0 0l5 5M7.188 2.239l.777 2.897M5.136 7.965l-2.898-.777M13.95 4.05l-2.122 2.122m-5.657 5.656l-2.12 2.122" /></svg></span> <span className="text-sm text-neutral-300">Right Click to Place/Interact</span></li>
                    </ul>
                 </div>
                 
                 <div className="bg-[#0A0A0B]/80 p-6 rounded-3xl border border-white/5">
                    <h3 className="text-sm uppercase tracking-widest text-neutral-500 font-bold mb-4">Inventory & UI</h3>
                    <ul className="space-y-4">
                      <li className="flex items-center gap-3"><span className="w-8 h-8 rounded-lg bg-white/10 flex items-center justify-center text-xs font-bold text-white border border-white/5">E</span> <span className="text-sm text-neutral-300">Open Inventory/Crafting</span></li>
                      <li className="flex items-center gap-3"><span className="w-8 h-8 rounded-lg bg-white/10 flex items-center justify-center text-xs font-bold text-white border border-white/5">Q</span> <span className="text-sm text-neutral-300">Toss Item</span></li>
                      <li className="flex items-center gap-3"><span className="w-8 h-8 rounded-lg bg-white/10 flex items-center justify-center text-xs font-bold text-white border border-white/5">1-9</span> <span className="text-sm text-neutral-300">Select Hotbar Slot</span></li>
                      <li className="flex items-center gap-3"><span className="w-8 h-8 rounded-lg bg-white/10 flex items-center justify-center text-xs font-bold text-white border border-white/5">↵</span> <span className="text-sm text-neutral-300">Open Global Chat</span></li>
                    </ul>
                 </div>
              </div>
              
              <button 
                onClick={() => setShowInstructions(false)}
                className="w-full bg-gradient-to-r from-blue-600 to-indigo-600 hover:from-blue-500 hover:to-indigo-500 text-white font-bold py-4 rounded-2xl transition-all shadow-[0_0_20px_rgba(37,99,235,0.3)] active:scale-[0.98] flex items-center justify-center gap-2"
              >
                Enter the World <ArrowRight size={20} />
              </button>
            </div>
          </div>
        )}
`;

code = code.replace(/\{\/\* Instructions Modal \*\/\}\n\s*\{showInstructions && \([\s\S]*?Start Playing\n\s*<\/button>\n\s*<\/div>\n\s*<\/div>\n\s*\)\}/, updatedInstructions);
fs.writeFileSync('src/App.tsx', code);

File: fix_ui.py
lines = open('src/App.tsx').read().split('\n')
for i in range(len(lines)):
    if "{/* Chest Modal Overlay */}" in lines[i]:
        insert = """
        {/* Ready Check Modal */}
        {readyCheck && (
           <div className="absolute inset-0 bg-black/80 backdrop-blur-md flex items-center justify-center z-50">
               <div className="bg-[#1A1A1E] border border-indigo-500/30 p-8 rounded-3xl shadow-2xl max-w-sm w-full flex flex-col items-center">
                   <Activity size={48} className="text-indigo-500 mb-4 animate-pulse" />
                   <h2 className="text-2xl font-black text-white tracking-tight mb-2 text-center">Dungeon Ready!</h2>
                   <p className="text-neutral-400 text-center mb-8 text-sm">Your party is ready for {readyCheck.instanceId}.</p>
                   <div className="flex gap-4 w-full">
                       <button onClick={() => { if (socketRef.current) socketRef.current.emit('decline_ready_check'); setReadyCheck(null); }} className="flex-1 bg-red-600/20 hover:bg-red-500 hover:text-white text-red-500 font-bold py-3 rounded-xl transition-all">Decline</button>
                       <button onClick={() => { if (socketRef.current) socketRef.current.emit('accept_ready_check'); }} className="flex-1 bg-indigo-600 hover:bg-indigo-500 text-white font-bold py-3 rounded-xl shadow-[0_0_20px_rgba(79,70,229,0.4)] transition-all">Accept</button>
                   </div>
               </div>
           </div>
        )}

        {/* Instances Modal */}
        {instancesOpen && (
           <div className="absolute inset-0 bg-black/60 backdrop-blur-sm flex items-center justify-center z-40" onClick={() => setInstancesOpen(false)}>
               <div className="bg-[#0A0A0B] border border-neutral-800 w-full max-w-3xl rounded-3xl overflow-hidden shadow-2xl flex flex-col" style={{ maxHeight: '80vh' }} onClick={e => e.stopPropagation()}>
                   <div className="bg-[#1A1A1E] p-4 flex justify-between items-center border-b border-neutral-800">
                       <h2 className="text-white font-bold flex items-center gap-2"><Globe size={18} className="text-indigo-500"/> Instance Finder</h2>
                       <button onClick={() => setInstancesOpen(false)} className="text-neutral-400 hover:text-white p-2">
                           <X size={20} />
                       </button>
                   </div>
                   <div className="p-6 overflow-y-auto">
                       <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
                           <div className="bg-[#151518] border border-neutral-800 rounded-2xl p-5 flex flex-col justify-between">
                               <div>
                                   <h3 className="text-xl font-black text-white mb-2">Linear Dungeon</h3>
                                   <p className="text-neutral-400 text-sm mb-4">A straightforward dark tunnel filled with dangerous mobs and a challenging boss encounter at the very end.</p>
                                   <div className="flex items-center gap-2 text-xs font-bold text-neutral-500 uppercase tracking-wider mb-6">
                                       <Activity size={14} className="text-red-500" /> Recommended: Party of 2+
                                   </div>
                               </div>
                               <button onClick={() => {
                                   if (socketRef.current) socketRef.current.emit('queue_instance', { instanceId: 'dungeon' });
                               }} className="w-full bg-indigo-600 hover:bg-indigo-500 text-white font-bold py-3 rounded-xl transition-all shadow-[0_0_15px_rgba(79,70,229,0.4)]">
                                   Queue for Dungeon
                               </button>
                           </div>
                       </div>
                   </div>
               </div>
           </div>
        )}
"""
        lines.insert(i, insert)
        break
open('src/App.tsx', 'w').write('\n'.join(lines))
